### PR TITLE
Show visual feedback when user needs to zoom to see selectable features

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10825

This PR adds an alert to show users when they need to zoom in to see the selectable CTN layers during the map step of creating or editing a component. The alert should only show when:
1. You are zoomed out beyond the level where we fetch data and show the CTN layers
2. You creating or editing a component and adding features with the map
3. You are not actively drawing component features

## Testing
**URL to test:** 

local or https://md-10825-zoom-alert--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Create a new project and go to the **Map** tab
4. Do not change the zoom of the map
5. Create a new component, fill out the form, and click **Continue**
6. When the map shows, notice that:
   - an alert shows in the top map bar telling you to zoom in to see selectable features
   - as you zoom in, you will reach a point where the CTN dashed lines or points layer appears and the alert disappears
7. Zoom back out to make the alert come back
8. Click on the draw tools in the top right to draw a line or point
9. When drawing, the alert should not show
10. You can try out the same steps with editing a component

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] ~Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js [hunk 1]
```diff
@@ -262,79 +264,95 @@ export default function TheMap({
...
-          createDispatch={createDispatch}
-          linkMode={linkMode}
-          setCursor={setCursor}
-          setIsDrawing={setIsDrawing}
+    <>
+      <MapAlertSnackbar
+        isOpen={shouldShowZoomAlert}
+        message="Zoom in to select map features"
+        severity="error"
+      />
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js:278]: This is the important part - the rest of the diff comes from me wrapping the **MapGL** component with a fragment.

File: moped-editor/src/views/projects/projectView/ProjectComponents/MapAlertSnackbar.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    top: "8px",
+  },
+}));
+
+const MapAlertSnackbar = ({
+  message,
+  severity,
+  isOpen,
+  snackbarProps,
+  alertProps,
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/MapAlertSnackbar.js:19]: Saw this in a video as a way to make sure we can pass props supported by the MUI pieces of this component in the future. Let me know what y'all think about this pattern. I think that it is cool but might be overkill.

**Review by patrickm02L — APPROVED** [review]
Functionality works well. The banner shows up clearly at appropriate zoom points. 

The placement of the banner is something I'm grappling with. I like it in the header, but it covers up longer Project Names. I could see the Banner in the map by the draw tool and the trash can.

**Video Overview**
- :05 Testing zoom points
- :34 Edit mode zooming
- :41 Reaction to the placement of the banner, alternative placement 
- 1:01 Creating a longer Project Name to see how the banner is displayed 
- 1:45 Testing banner functionality in the map with longer Project Name
- 2:00 Another pitch to move the banner next to the tool and trash can on the map 

https://user-images.githubusercontent.com/5566727/211041131-ae9fbbaf-c0be-4965-a557-816dd359ac43.mp4

**Review by roseeichelmann — APPROVED** [review]
Love it! This feature is such a huge improvement to the UI imo and it's great that you went ahead and made an alert snackbar component out of this that can be reused for other potential map alerts we may come to need. I have to agree with Patrick about the placement of the snackbar, could be interesting to try it elsewhere on the map like by the drawing tools as he suggested. Also just have to shoutout your code, it's so clean and readable. Snaps all around 🫰🏻🫰🏻🫰🏻

**Review by johnclary — COMMENTED** [review]
thanks @patrickm02L for providing the test case of very long project name.

@mddilley my preference would be to have the alert banner flow inline with the other toolbar content, such that it was always to the right of the status badge. i realize that wasn't obvious in [the spec](https://camo.githubusercontent.com/09676d06080977660dc500d4638ace008a19ba8604c95f08ffdaff77de20a4af/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3563616636373662656361643131353331636334313763622f66343135373638642d626366372d343038362d626232302d616366326330303236343632).

i think what you have is good, but i'd rather not cover up the project name if you feel it's trivial to move the alert out the snackbar and into normal flow. otherwise i'll make a backlog issue and we can revisit later.

**Comment by mddilley** [comment]
@johnclary I'll work those updates into this PR for sure. I'll work on that today so we keep the project name in view when showing the alert. 🙏

**Comment by johnclary** [comment]
thanks @mddilley!

**Comment by mddilley** [comment]
@johnclary Just pushed some updates that refactor that alert with a reducer since the state needed to move up a level to hide/show the alert. Could probably clean up the code a bit but wanted to see what you thought first. 🙏

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/useToolbarErrorMessage.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { useReducer } from "react";
```
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/useToolbarErrorMessage.js:1]: TIL `useReducer`. Thanks for that. I think this is a great managing complex state without getting something like `redux` involved. Super nice.

**Review by frankhereford — APPROVED** [review]
I was looking at another one of your PRs earlier this evening, and I saw this alert come up, and I had a screenshot all ready to put in my review saying how much I loved this. I'm glad I peeked the remaining PRs to review, and saw this one was just a few down the list. 

This is fantastic, and I agree with Rose - reading your work reminds me what I strive for in my work -- it's just so clean. I worked through the testing and looked over the diff, I think this is ready to set sail. ⛵️

**Review by johnclary — APPROVED** [review]
love the `useEffect` + reducer dispatch - looks great and works great. thanks @mddilley!!

🚢 🚢 🚢 🚢 🚢 🚢 🚢

**Pull request merged**